src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -26,17 +34,26 @@ public class ContentResult : ActionResult
         {
             var response = context.HttpContext.Response;
 
-            MediaTypeHeaderValue contentTypeHeader;
-            if (string.IsNullOrEmpty(ContentType))
+            var contentTypeHeader = ContentType;
+            Encoding encoding;
+            if (contentTypeHeader == null)
             {
-                contentTypeHeader = new MediaTypeHeaderValue("text/plain");
+                contentTypeHeader = DefaultContentType;
+                encoding = DefaultContentType.Encoding;
```

Comment:
yeah, right...its UTF8, but we are not specifying the BOM setting as it has no effect in this case, for example when writing content using StreamWriter it writes out the BOM if the encoding requires it, but we are using the following `response.WriteAsync` api which does not use StreamWriter...

https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http.Abstractions/HttpResponseWritingExtensions.cs#L39